Assets/Scripts/UI/InGame/HUD/CompassController.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Assets.Scripts.UI.InGame.HUD
+{
+    public class CompassController : UiElement
+    {
+        [Header("Customization")]
+        public RawImage compassImage;
+        [Header("Settings")]
+        public float maxDistance; 
+        public GameObject iconPrefab;
+        public GameObject markersParent;
+        [Header("Markers")]
+        public List<CompassMarker> compassMarkers = new List<CompassMarker>();
+        [HideInInspector]public bool compassMode;
+        [HideInInspector]public Transform cam; //Is declared by the IN_GAME_MAIN_CAMERA script
+
+
+
+        float compassUnit;
+
+        void Start()
+        {
+            //1 compass unit equals 1 degree angle
+            compassUnit = compassImage.rectTransform.rect.width / 360f;
+            maxDistance = 150f; //minimap max distance
+        }
+
+        void Update()
+        {
+            if(compassMode)
```

Comment:
Invert if statement to make this code more readable.

`if(!CompassMode) return;`
Now the remaining 20 lines of code, don't have to be within that if statement